cli/azd/pkg/commands/pipeline/azdo_provider.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package pipeline
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+)
+
+// AzdoHubScmProvider implements ScmProvider using Azure DevOps as the provider
+// for source control manager.
+type AzdoHubScmProvider struct {
+}
+
+// ***  subareaProvider implementation ******
+
+// requiredTools return the list of external tools required by
+// Azure DevOps provider during its execution.
+func (p *AzdoHubScmProvider) requiredTools() []tools.ExternalTool {
+	return []tools.ExternalTool{}
+}
+
+// preConfigureCheck check the current state of external tools and any
+// other dependency to be as expected for execution.
+func (p *AzdoHubScmProvider) preConfigureCheck(ctx context.Context, console input.Console) error {
+	return nil
```

Comment:
just wondering, should we return something like `errors.New("not implemented")` for now from these stubbed out methods?